code/game/machinery/spaceheater.dm

```diff
@@ -234,6 +235,159 @@
 				cell = null
 				. = TRUE
 
+///For use with heating reagents in a ghetto way
+/obj/machinery/space_heater/improvised_chem_heater
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "sheater-off"
+	name = "Improvised chem heater"
+	desc = "A space heater hacked to reroute heating to a water bath on the top."
+	panel_open = TRUE //This is always open
+	var/obj/item/reagent_containers/beaker = null
+	var/chem_heating_power = 1
+
+/obj/machinery/space_heater/improvised_chem_heater/process(delta_time)
+	if(!on || !is_operational)
+		if (on) // If it's broken, turn it off too
+			on = FALSE
+		return PROCESS_KILL
+
+	if(!cell || cell.charge < 0)
+		on = FALSE
+		return PROCESS_KILL
```

Comment:
```suggestion
	if(!on)
		return PROCESS_KILL

	if(!is_operational || !cell || cell.charge <= 0)
		on = FALSE
		return PROCESS_KILL
```
* Less redundant checks.
* You probably don't want the process to continue with a charge of zero.